src/poliastro/contrib/extract_czml.py

```diff
@@ -0,0 +1,279 @@
+        i: int
+            Index of referenced orbit
+        """
+        h = (self.end_epoch - self.orbits[i][2]).to(u.second) / self.orbits[i][1]
+
+        for k in range(self.orbits[i][1] + 2):
+            cords = (
+                self.orbits[i][0]
+                .represent_as(CartesianRepresentation)
+                .xyz.to(u.meter)
+                .value
+            )
+            cords = np.insert(cords, 0, h.value * k, axis=0)
+
+            self.czml[i]["position"]["cartesian"] += cords.tolist()
+            self.orbits[i][0] = self.orbits[i][0].propagate(h)
+
+    def __init_czml__(self):
+        """
+        Only called at the initialization of the extractor
+        Builds packets.
+        """
+
+        self.parse_dict_tuples(
+            [-1], [("id", "document"), ("name", "simple"), ("version", "1.0")]
+        )
+        self.parse_dict_tuples(
+            [-1, "clock"],
+            [
+                ("interval", self.start_epoch.value + "/" + self.end_epoch.value),
+                ("currentTime", self.start_epoch.value),
+                ("multiplier", 60),
+                ("range", "LOOP_STOP"),
+                ("step", "SYSTEM_CLOCK_MULTIPLIER"),
+            ],
+        )
+
+    def change_id_params(self, i, id=None, name=None, description=None):
+        """
+        Change the id parameters.
+
+        Parameters
+        i : int
+            Referred body (count starts at i)
+        id: str
+            Set orbit id
+        name: str
+            Set orbit name
+        description: str
+            Set orbit description
+        """
+        if id is not None:
+            self.parse_dict_tuples([i], [("id", id)])
+        if name is not None:
+            self.parse_dict_tuples([i], [("name", name)])
+        if description is not None:
+            self.parse_dict_tuples([i], [("description", description)])
+
+    def change_path_params(
+        self, i, pixel_offset=None, color=None, width=None, show=None
+    ):
+        """
+        Changes the path parameters.
+
+        Parameters
+        ----------
+        i : int
+            Referred body (count starts at 1)
+        pixel_offset: list (int)
+            The pixel offset (up and right)
+        color: list (int)
+            Rgba path color
+        width: int
+            Path width
+        show: bool
+            Indicates whether the path is visible
+        """
+        if pixel_offset is not None:
+            self.parse_dict_tuples(
+                [i, "label", "pixelOffset"], [("cartesian2", pixel_offset)]
+            )
+        if color is not None:
+            self.parse_dict_tuples(
+                [i, "path", "material", "solidColor", "color"], [("rgba", color)]
+            )
+        if width is not None:
+            self.parse_dict_tuples([i, "path"], [("width", width)])
+        if show is not None:
+            self.parse_dict_tuples([i, "path", "show"], [("boolean", show)])
+
+    def change_label_params(
```

Comment:
I tried this function and seemed to change the labels of all orbits, by the way.

Reply:
@Juanlu001 I just looked into it and you're right. It was my mistake for using ``copy()`` which creates a shallow copy of the ``DEFAULTS``. Anyway, I'll push the fix a bit later along with other changes you proposed.